tests/src/providers/testqgspostgresconn.cpp

```diff
@@ -38,6 +40,36 @@ class TestQgsPostgresConn: public QObject
       QCOMPARE( QgsPostgresConn::quotedValue( "b \"c' \\x" ), QString( "E'b \"c'' \\\\x'" ) );
     }
 
+    void quotedValueDatetime()
+    {
+      QCOMPARE( QgsPostgresConn::quotedValue( QDateTime::fromString( "2020-05-07 17:56:00", "yyyy-MM-dd HH:mm:ss" ) ), QString( "'2020-05-07T17:56:00'" ) );
+
+      QgsFields fields;
+      QgsField f;
+      f.setName( "ts" );
+      f.setType( QVariant::DateTime );
+      f.setTypeName( "timestamp" );
+      fields.append( f );
+      QgsField f2;
+      f2.setName( "pk" );
+      f2.setType( QVariant::ULongLong );
+      fields.append( f2 );
+
+      QList<int> pkAttrs;
+      pkAttrs.append( 0 );
+      pkAttrs.append( 1 );
+
+      QgsPostgresSharedData *shared = new QgsPostgresSharedData;
+      std::shared_ptr<QgsPostgresSharedData> sharedPtr( shared );
```

Comment:
Either allocate on stack or use a unique_ptr.

Reply:
is it better?

Reply:
Unfortunately not, your new version has a memory leak: you should probably allocate on stack or use a unique_ptr.